app/models/miq_expression.rb

```diff
@@ -1775,15 +1775,9 @@ def self.atom_error(field, operator, value)
                                                                                     :error_message => err.message}
       end
       return false
-    when :ruby
-      return _("Ruby Script must not be blank") if value.blank?
-      return false
```

Comment:
@yrudman I'm late to this PR but it's hard to follow when a single commit is doing more than one thing.

I would suggest 

commit 1: add tests.
commit 2: drop when "ruby" # TODO above,  here, + early return
commit 3: reduce the calls to `operator.downcase`